### PR TITLE
OCD-3849 - Implement email try/resend feature

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java [hunk 1]
```diff
@@ -69,6 +69,10 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private ChplEmailFactory chplEmailFactory;
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java:60]: Could you move these fields into an Autowired constructor? Just as a refactor to keep up with our current coding style

File: chpl/chpl-api/src/main/resources/log4j2-xinclude-file-appenders-json.xml [hunk 2]
```diff
@@ -979,4 +979,24 @@
...
         </DefaultRolloverStrategy>
     </RollingFile>
+    <RollingFile name="sendEmailJobJson"
+        fileName="${logDir}/scheduler/sendEmailJob.json"
+        filePattern="${logDir}/scheduler/history/sendEmailJob-%d{yyyy-MM-dd}.json"
+        filePermissions="rw-rw-r--">
+        <JsonLayout compact="true" eventEol="true" properties="true" stacktraceAsString="true">
+            <KeyValuePair key="service" value="sendEmailJob" />
+            <KeyValuePair key="dd.trace_id" value="%X{dd.trace_id}" />
+            <KeyValuePair key="dd.span_id" value="%X{dd.trace_id}" />
```
Comment (kekey1) [inline chpl/chpl-api/src/main/resources/log4j2-xinclude-file-appenders-json.xml]: Curious about this line. I assume it was copy/pasted from elsewhere, but should the "dd.span_id" have a value of "dd.span_id" or is it correct that both fields have the value trace_id?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/email/ChplEmailFactory.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.quartz.Scheduler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.scheduler.ChplSchedulerReference;
+
+@Component
+public class ChplEmailFactory {
+    //private SchedulerManager schedulerManager;
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/email/ChplEmailFactory.java]: Remove commented out references to schedulerManager

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/SendEmailJob.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                    .map(file -> copyTempFileToPermanentLocation(file))
+                    .collect(Collectors.toList());
+        } else {
+            return message.getFileAttachments();
+        }
+
+    }
+
+    private File copyTempFileToPermanentLocation(File originalFile) {
+        Path newPath = Paths.get(env.getProperty("downloadFolderPath") + File.separator + originalFile.getName());
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/SendEmailJob.java]: Is this the directory we want to use? Or would we rather have a new and distinct folder we add to the machines that we use for this?

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/SendEmailJob.java (hunk 4 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/SendEmailJob.java]: I see this was changed but I wonder if you could make the folder name be a property or constant at least? Also will need a checklist item to create the directory for all environments.

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**